corehq/apps/userreports/reports/filters/specs.py

```diff
@@ -19,21 +23,11 @@
     QuarterFilterValue,
     LocationDrilldownFilterValue, MultiFieldChoiceListFilterValue)
 from corehq.apps.userreports.specs import TypeProperty
+from django.conf import settings
+from django.utils.module_loading import import_string
 
 
-class ReportFilter(JsonObject):
```

Comment:
I think it's a bit nicer to have `class.name` instead of `class["name"]`. It's also nice that `JsonObject` has static typing. Was there a particular reason for this refactor? Maybe a namedtuple would be a good compromise?

Reply:
It was confusing to have this class because there are already UCR filter classes (eg. LocationDrilldownFilter etc), and it's confusing when you read the code what class was being passed. What this class was doing was essentially jsonobject-ing the common attributes of the filter specs. By removing it and directly passing the whole filter spec it's now possible to access other spec attributes (that are required in some cases like the ancestor-filter). `namedtuple` will also not address the extra attribute.